### ISSUE TITLE
How to prevent gradient exploding?

### ISSUE THREAD

**Issue by 103yiran** [foolwood/DaSiamRPN#47]
Thank you for your contributions!
It seems that the Correlation operation leads to huge gradients. As the paper says that your init_lr is 1e-2. Can you tell me the parameter initialization you use? Is the xavier OK for this, or can you share other tricks to avoid gradient exploding?

**Comment by ucasqcz** [comment]
> Thank you for your contributions!
> It seems that the Correlation operation leads to huge gradients. As the paper says that your init_lr is 1e-2. Can you tell me the parameter initialization you use? Is the xavier OK for this, or can you share other tricks to avoid gradient exploding?

in my experiment, normal distribution with delta 0.005 is able to avoid NAN, for correlation operation,  siamfc also uses a conv layer init with 1e-3 to scale the output correlation map, i do not know the solution of the author, by the way, the pre-trained model may do some favor to the performance, and lr warm-up is worthy to try.
